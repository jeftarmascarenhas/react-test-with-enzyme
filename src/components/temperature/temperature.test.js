/* eslint-disable no-undef */
import React from "react";

import Temperature from "./index";

describe("<Temperature />", () => {
  it("renders correctly", () => {
    const wrapper = render(
      <Temperature temp={10} city="Recife" toggleForecast={() => {}} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("formats temp correctly", () => {
    const wrapper = mount(
      <Temperature temp={10} city="Agua Preta" toggleForecast={() => {}} />
    );
    const text = wrapper.find("h2").text();
    expect(text).toEqual("10Â°c");
  });

  it("formats city correctly", () => {
    const wrapper = mount(
      <Temperature temp={10} city="Recife" toggleForecast={() => {}} />
    );
    const text = wrapper.find("h3").text();
    expect(text).toEqual("Recife");
  });

  it("calls toggleForecast on click", () => {
    const spy = sinon.spy();
    const wrapper = mount(
      <Temperature temp={10} city="Olinda" toggleForecast={spy} />
    );

    wrapper
      .find("div")
      .first()
      .simulate("click");
    expect(spy.calledOnce).toBe(true);
  });
});
