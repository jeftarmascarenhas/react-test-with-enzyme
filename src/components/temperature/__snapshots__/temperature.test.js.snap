// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Temperature /> renders correctly 1`] = `
.c0 {
  background-color: #111111;
  box-shadow: 1px 1px 2px rgba(0,0,0,0.1);
  padding: 10px;
  border-radius: 2px;
}

.c1 {
  font-size: 20px;
  color: #fff;
}

.c2 {
  font-size: 16px;
  color: #fff;
}

<div
  class="c0"
>
  <h2
    class="c1"
  >
    10Â°c
  </h2>
  <h3
    class="c2"
  >
    Recife
  </h3>
</div>
`;
